eclair-core/src/main/scala/fr/acinq/eclair/MilliSatoshi.scala

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2019 ACINQ SAS
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fr.acinq.eclair
+
+import fr.acinq.bitcoin.{Btc, BtcAmount, MilliBtc, Satoshi, btc2satoshi, millibtc2satoshi}
+
+/**
+ * Created by t-bast on 22/08/2019.
+ */
+
+/**
+ * One MilliSatoshi is a thousand of a Satoshi, the smallest unit usable in bitcoin
+ */
+case class MilliSatoshi(private val underlying: Long) extends Ordered[MilliSatoshi] {
+
+  // @formatter:off
+  def +(other: MilliSatoshi) = MilliSatoshi(underlying + other.underlying)
+  def +(other: BtcAmount) = MilliSatoshi(underlying + other.toMilliSatoshi.underlying)
+  def -(other: MilliSatoshi) = MilliSatoshi(underlying - other.underlying)
+  def -(other: BtcAmount) = MilliSatoshi(underlying - other.toMilliSatoshi.underlying)
+  def *(m: Long) = MilliSatoshi(underlying * m)
+  def *(m: Double) = MilliSatoshi((underlying * m).toLong)
+  def /(d: Long) = MilliSatoshi(underlying / d)
+  def unary_-() = MilliSatoshi(-underlying)
+  override def compare(other: MilliSatoshi): Int = if (underlying == other.underlying) 0 else if (underlying < other.underlying) -1 else 1
+  // Since BtcAmount is a sealed trait that MilliSatoshi cannot extend, we need to redefine comparison operators.
+  def compare(other: BtcAmount): Int = {
+    val otherMsat = other.toMilliSatoshi.underlying
+    if (underlying == otherMsat) 0 else if (underlying < otherMsat) -1 else 1
```

Comment:
Not a strong opinion but i would try to reuse the code as much as possible. For example here you could reuse the `compare(other: MilliSatoshi)` implementation instead of re-writing again the comparison logic, even the method above could use the `compareTo` offered by the java api.